src/components/Modal/Modal.js

```diff
@@ -0,0 +1,64 @@
+import { useEffect, useRef } from 'react';
+import './Modal.css';
+
+const Modal = ({ showModal, handleModalClose, deleteItem, item }) => {
+  const toggleModalClassName = showModal ? 'display-block' : 'display-none';
+
+  const cancelRef = useRef();
+  const deleteRef = useRef();
+
+  useEffect(() => {
+    if (showModal) {
+      // when modal opens, add eventListeners and put initial focus on "No, Cancel"
+      document.addEventListener('keydown', handleKeyEvents);
+      cancelRef.current.focus();
+    } else {
+      document.removeEventListener('keydown', handleKeyEvents);
+    }
+  }, [showModal]);
+
+  const handleKeyEvents = (e) => {
+    // close modal if user hits Escape (27)
+    if (e.keyCode === 27) {
+      handleModalClose();
+    }
+
+    // this keeps keyboard focus within modal. if user hits Tab (9)
+    if (e.keyCode === 9) {
+      // ...and NOT shift while on Delete, put focus on Cancel
+      if (!e.shiftKey && document.activeElement === deleteRef.current) {
+        cancelRef.current.focus();
+        return e.preventDefault();
+      }
+
+      // ...and shift while on Cancel, put focus on Delete
+      if (e.shiftKey && document.activeElement === cancelRef.current) {
+        deleteRef.current.focus();
+        return e.preventDefault();
+      }
+    }
+  };
```

Comment:
Really impressive stuff here, everything in this useEffect is new to me conceptually but you really commented it well and made it easier to understand.

Reply:
Yes! If you comment this code out and use a keyboard to open the modal, if you start tabbing or shift/tabbing into the modal, the keyboard focus will start leaving the modal into the shopping list behind it.